specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/examples/expressRouteProviderPort.json

```diff
@@ -0,0 +1,239 @@
+{
+	"swagger": "2.0",
+	"info": {
+		"title": "NetworkManagementClient",
+		"description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+		"version": "2021-08-01"
+	},
+	"host": "management.azure.com",
+	"schemes": [
+		"https"
+	],
+	"consumes": [
+		"application/json"
+	],
+	"produces": [
+		"application/json"
+	],
+	"security": [
+		{
+			"azure_auth": [
+				"user_impersonation"
+			]
+		}
+	],
+	"securityDefinitions": {
+    "azure_auth": {
+			"type": "oauth2",
+			"authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+			"flow": "implicit",
+			"description": "Azure Active Directory OAuth2 Flow.",
+			"scopes": {
+				"user_impersonation": "impersonate your user account"
+			}
+		}
+	},
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/expressRouteProviderPorts": {
+      "get": {
+        "tags": [
+          "ExpressRouteProviderPorts"
+        ],
+        "operationId": "ExpressRouteProviderPorts_List",
+        "description": "Retrieves all the ExpressRouteProviderPorts in a subscription.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of ExpressRouteProviderPort resources. If there are no ports for the provider, an empty list is returned.",
+            "schema": {
+              "$ref": "#/definitions/ExpressRouteProviderPortListResult"
+            }
+          },
+          "default": { 
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ExpressRouteProviderPortList": {
+            "$ref": "./examples/ExpressRouteProviderPortList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/expressRouteProviderPorts": {
```

Comment:
> "/subscriptions/{subscriptionId}/providers/Microsoft.Network/expressRouteProviderPorts" [](http://example.com/codeflow?start=3&length=88)

Your path is invalid you have duplicate paths. Will need to check what is the best option in this case, not sure if a query string is the best option, could be but want to confirm and get back to you.

Reply:
Can you use odata to add a filter instead of passing a query string? [https://armwiki.azurewebsites.net/rpaas/odata_filtering.html?q=odata](https://armwiki.azurewebsites.net/rpaas/odata_filtering.html?q=odata)

Here's an example of our RP using it: [https://github.com/Azure/azure-rest-api-specs/blob/12aa4dd9607f82bdc1087f24b91cec7d5c4499c2/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json#L353](https://github.com/Azure/azure-rest-api-specs/blob/12aa4dd9607f82bdc1087f24b91cec7d5c4499c2/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json#L353)

Reply:
done